src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -193,16 +202,20 @@ private void CacheInstanceDiscoveryMetadata(InstanceDiscoveryMetadataEntry metad
 
         private async Task<Uri> BuildAuthorityWithRegionAsync(Uri canonicalAuthority, RequestContext requestContext)
         {
-            string region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+            if (s_region.IsNullOrEmpty())
+            {
+                s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+            }
+            
             var builder = new UriBuilder(canonicalAuthority);
 
             if (KnownMetadataProvider.IsPublicEnvironment(canonicalAuthority.Host))
             {
-                builder.Host = $"{region}.login.microsoft.com";
+                builder.Host = $"{s_region}.login.microsoft.com";
             }
```

Comment:
do we have to worry about sovereign clouds in this scenario?

Reply:
No, currently ests-r support only public clouds and in case it is anything other than public cloud we are appending the region to the authority. This is confirmed with the ests team.